alloy-core/src/main/java/org/starchartlabs/alloy/core/Strings.java

```diff
@@ -328,4 +328,29 @@ private static boolean validSurrogatePairAt(CharSequence string, int index) {
                 && Character.isHighSurrogate(string.charAt(index))
                 && Character.isLowSurrogate(string.charAt(index + 1));
     }
+
```

Comment:
Is this duplicated here for dependency reasons (i.e. should a user be able to get `Strings` somehow without pulling in `Preconditions`)? Otherwise I feel like this should just be using `Preconditions.checkArgument` still

Reply:
This was explained to me in person as being to avoid a cyclical dependency, as Preconditions relies on String, and this was the only spot where it was the other way around